src/features/auth/SignupComplete.js

```diff
@@ -0,0 +1,197 @@
+import { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+import { auth, storage } from "../../firebase";
+import {
+  Alert,
+  Button,
+  Col,
+  Container,
+  Form,
+  InputGroup,
+  Row,
+} from "react-bootstrap";
+import { Navigation } from "../../components";
+import { isUsernameExist } from "../../services/auth";
+import { fetchUserById, signupComplete } from "./authSlice";
+import useDocumentTitle from "../../hooks/useDocumentTitle";
+import styles from "./SignupComplete.module.css";
+
+const MAX_AVATAR_SIZE = 500 * 1024;
+
+const SignupComplete = () => {
+  const { uid, username } = useSelector((state) => state.auth);
+  const [isDisabled, setDisabled] = useState(true);
+  const [alertDisplay, setAlertDisplay] = useState(false);
+  const [error, setError] = useState("");
+  const [formData, setFormData] = useState({
+    displayName: "",
+    username: "",
+    avatar: "",
+    bio: "",
+  });
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  useDocumentTitle("Create your account on RabiBook");
+
+  useEffect(() => {
+    const unsubscribe = auth.onAuthStateChanged((user) => {
+      if (!user) {
+        return navigate("/home");
+      }
+      const { uid } = user;
+      dispatch(fetchUserById(uid));
+      if (username) {
+        navigate("/");
+      }
+    });
+    return () => unsubscribe();
+  }, [dispatch, navigate, username]);
+
+  const handleFileInput = async (e) => {
+    if (e.target.files[0]?.size > MAX_AVATAR_SIZE) {
+      setError(`File size should be less than ${MAX_AVATAR_SIZE / 1024} KB`);
+      setAlertDisplay(true);
+      setDisabled(true);
+      return;
```

Comment:
Looks like this part is used in many places.

So why not create a upload component wrapper which will manage all of this.